src/modules/flow/json/json.c

```diff
@@ -106,83 +105,103 @@ json_object_search_for_token(struct sol_json_scanner *scanner, const char *key,
-                new_blob);
+                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__ARRAY, new_blob);
             sol_blob_unref(new_blob);
             return r;
-        case SOL_JSON_TYPE_TRUE:
-            return sol_flow_send_boolean_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
-                true);
-        case SOL_JSON_TYPE_FALSE:
-            return sol_flow_send_boolean_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
-                false);
-        case SOL_JSON_TYPE_NULL:
-            return sol_flow_send_empty_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__NULL);
-        case SOL_JSON_TYPE_STRING:
-            sol_json_token_remove_quotes(&value);
-            slice = sol_json_token_to_slice(&value);
-            return sol_flow_send_string_slice_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__STRING,
-                slice);
-        case SOL_JSON_TYPE_NUMBER:
-            r = sol_json_token_get_double(&value, &value_float.val);
-            SOL_INT_CHECK(r, < 0, r);
-            r = sol_flow_send_drange_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__FLOAT,
-                &value_float);
-            SOL_INT_CHECK(r, < 0, r);
-
-            r = sol_json_token_get_int32(&value, &value_int.val);
-            if (r == -EINVAL) /* Not an integer number */
-                return 0;
-            SOL_INT_CHECK(r, < 0, r);
-            return sol_flow_send_irange_packet(node,
-                SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__INT,
-                &value_int);
-        default:
-            slice = sol_json_token_to_slice(&value);
-            return sol_flow_send_error_packet(node, EINVAL,
-                "JSON Object value %.*s, pointed by key %s is invalid",
-                SOL_STR_SLICE_PRINT(slice), mdata->key);
         }
+        new_blob = create_sub_json(json, scanner, token,
+            SOL_JSON_TYPE_ARRAY_END);
+        SOL_NULL_CHECK(new_blob, -errno);
+        r = sol_flow_send_json_array_packet(node,
+            SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__ARRAY,
+            new_blob);
+        sol_blob_unref(new_blob);
+        return r;
+    case SOL_JSON_TYPE_TRUE:
+        return sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
+            true);
+    case SOL_JSON_TYPE_FALSE:
+        return sol_flow_send_boolean_packet(node,
+            SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__BOOLEAN,
+            false);
+    case SOL_JSON_TYPE_NULL:
+        return sol_flow_send_empty_packet(node,
+            SOL_FLOW_NODE_TYPE_JSON_OBJECT_GET_KEY__OUT__NULL);
+    case SOL_JSON_TYPE_STRING:
+        r = sol_json_token_get_unescaped_string(token, &buffer);
+        SOL_INT_CHECK(r, < 0, r);
+        r = sol_flow_send_string_slice_packet(node,
```

Comment:
you could use `get_unescaped_string_copy()` and then use `sol_flow_send_string_take_packet()`, so you reduce allocations if it was escaped (the `take` variant will handle the pointer ownership to the packet).